### ISSUE TITLE
Refactor Helm YAML templates with _helpers.tpl

### ISSUE THREAD

**Issue by jiacheliu3** [Alluxio/alluxio#10016]
**Summary**
A few parts of the Helm YAML templates can be refactored using Helm's `_helpers.tpl`.

1. There are duplicate boolean flags duplicated in `alluxio-master.yaml` and `alluxio-format-master.yaml` for the Helm templates. The are better moved to a single place.

```
{{- $masterCount := int .Values.resources.masterCount }}
{{- $isSingleMaster := eq $masterCount 1 }}
{{- $isEmbedded := (eq .Values.journal.type "EMBEDDED") }}
{{- $isHaEmbedded := and $isEmbedded (gt $masterCount 1) }}
{{- $isUfsLocal := and (eq .Values.journal.type "UFS") (eq .Values.journal.ufsType "local") }}
{{- $isSingleUfsLocal := and $isUfsLocal $isSingleMaster }}
{{- $needJournalVolume := or $isEmbedded $isUfsLocal }}
```
2. The duplicate sections like `Volumes` and `VolumeMounts` can be templated and moved to `_helpers.tpl`. Now the same sections are duplicated in `alluxio-master.yaml` and `alluxio-format-master.yaml`.

**Urgency**
The duplicate sections are hard to maintain. Given the fact that the Helm YAML templates are actively developed, we should consider getting this refactored.

**Comment by yuzhu** [comment]
@jiacheliu3 has this been addressed by #10003?

**Comment by jiacheliu3** [comment]
Closed as this has been addressed in https://github.com/Alluxio/alluxio/pull/10686
